ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -231,6 +268,57 @@ private HelixBootstrapUpgradeUtil(String hardwareLayoutPath, String partitionLay
     }
   }
 
+  /**
+   * Create HelixPropertyStore for all datacenters. Also it checks if the zNode exists, if not the method will create a
+   * new zNode for PartitionOverride.
+   */
+  private void createHelixPropertyStoreForAllDCs() {
+    Properties storeProps = new Properties();
+    storeProps.setProperty("helix.property.store.root.path", "/" + this.clusterName);
+    HelixPropertyStoreConfig propertyStoreConfig = new HelixPropertyStoreConfig(new VerifiableProperties(storeProps));
+    for (Map.Entry<String, ClusterMapUtils.DcZkInfo> entry : dataCenterToZkAddress.entrySet()) {
+      ZkClient zkClient = new ZkClient(entry.getValue().getZkConnectStr(), propertyStoreConfig.zkClientSessionTimeoutMs,
+          propertyStoreConfig.zkClientConnectionTimeoutMs, new ZNRecordSerializer());
+      List<String> subscribedPaths = Collections.singletonList(propertyStoreConfig.rootPath);
+      HelixPropertyStore<ZNRecord> helixPropertyStore =
+          new ZkHelixPropertyStore<>(new ZkBaseDataAccessor<>(zkClient), propertyStoreConfig.rootPath, subscribedPaths);
```

Comment:
In this situation, you can pass in `null` to `subscribedPaths` since you are not using the listener functionality.